spec/models/solidus_pay_tomorrow/payment_source_spec.rb

```diff
@@ -43,4 +43,35 @@
       end
     end
   end
+
+  describe "#can_credit?" do
+    let(:payment) {
+      create(:payment, order: order, payment_method: payment_method, source: payment_source,
+        state: :completed, response_code: payment_source.application_token,
+        amount: order.total)
+    }
+
+    context 'when a refund already exists for the payment' do
+      before do
+        create(:refund, payment: payment, amount: payment.amount)
+      end
+
+      it 'returns false' do
+        expect(payment_source).not_to be_can_credit(payment)
+      end
+    end
+
+    context 'when payment is in state - completed' do
+      it 'returns true' do
+        expect(payment_source).to be_can_credit(payment)
+      end
+    end
+
+    context 'when payment is in state - pending' do
+      it 'returns true' do
+        payment.update!(state: 'pending')
```

Comment:
Instead of running this update! what do you think adding a payment_state var?
e.g.

```
let(:payment_state) { :complete }
let(:payment) {
      create(:payment, order: order, payment_method: payment_method, source: payment_source,
        state: payment_state, response_code: payment_source.application_token,
        amount: order.total)
    }
    ....
    ....
    
context 'when payment is in state - pending' do
  let(:payment_state) { :pending }
  ...
end
```

In this way, the test will be faster and we'll avoid to hit the database.

Reply:
Sure

Reply:
Done in https://github.com/nebulab/solidus_pay_tomorrow/pull/18/commits/c041363750bea16aa7501b90bb416eb066a4ffa0